### ISSUE TITLE
Broken synthaxe highlight since update

### ISSUE THREAD

**Issue by Kaauw** [samuelcolvin/jinjahtml-vscode#107]
Dear Team,

I think an update broke the synthaxe highlight. All braces are now colorized, not only jinja2 braces.
With this example : 

![image](https://user-images.githubusercontent.com/7242711/167582932-80af6112-feac-4a8c-b812-e08ce3c4d8b0.png)

As raw if you want to test : 

```jinja2
{% if device_family == "SRX" %}
security {
  apply-flags omit;
  forwarding-options {
    family {
      inet6 {
        mode packet-based;
      }
      mpls {
        mode packet-based;
      }
      iso {
        mode packet-based;
      }
    }
  }
}
{% endif %}
```

Thanks,

Benjamin

**Comment by samuelcolvin** [comment]
An update to this extension or to vscode.

**Comment by samuelcolvin** [comment]
humm, seems to be broken for me too. Seems like something must have changed in vscode.

Happy to accept a PR if someone can find a fix, but since I don't use vscode day-to-day, I can't devote the time to fix this myself.

**Comment by samuelcolvin** [comment]
I've been looking at this for some time. It  seems vscode somehow changes in how it interpreted grammars.

I tried  the other "Jinja" extension which hasn't been updated for ages and it seems to be broken in the same way.

Seems like it might involve rewriting the grammar completely to get it to work.

@nathan-barrett do you have any idea what might have changed in VS code to cause this to break?

**Comment by samuelcolvin** [comment]
See https://github.com/microsoft/vscode/issues/158379, I'm waiting for a respond from there.

**Comment by samuelcolvin** [comment]
Also worth noting that for me only one brace is highlighted. I tried changing the macro in the syntax definition, but nothing seemed to work.

<img width="210" alt="image" src="https://user-images.githubusercontent.com/4039449/185171463-c8c223be-bb68-4038-8ff4-86d18a5acd00.png">

**Comment by nathan-barrett** [comment]
Weird, let me take a look this afternoon and see if I can shake anything loose! thanks @samuelcolvin

**Comment by samuelcolvin** [comment]
https://github.com/microsoft/vscode/issues/158379

> @RedCMD @thqby Thanks a lot for helping out! heart
> 
> @samuelcolvin The issue [samuelcolvin/jinjahtml-vscode#107](https://github.com/samuelcolvin/jinjahtml-vscode/issues/107) is caused by us [enabling bracket pair colorization by default](https://code.visualstudio.com/updates/v1_67#_bracket-pair-colorization-enabled-by-default) in 1.67 which shipped early May 2022. @Kaauw can get the previous behavior by configuring `"editor.bracketPairColorization.enabled": false`

**Comment by cedsana** [comment]
Disabling `bracketPairColorization` does not fix the issue on my end, I still don't get any color for the jinja brackets/variables.

**Comment by RedCMD** [comment]
if you're getting no syntax highlighting at all
that sounds like a separate issue

can you send a screenshot
and a copy of your code

**Comment by dromer** [comment]
Seeing something similar happen in `.cpp.j2`:

Certain single-bracket enclosures get highlighted:
![](https://i.imgur.com/MyvjZfc.png)

And here we see perfectly fine jinja code being missed (because it's enclosed within such an identified block):
![](https://i.imgur.com/SIM6RKV.png)

I use cpp extensively with jinja, so this makes the plugin quite useless at the moment (also I miss not being able to lint the code, even though it's quirky and limited).

**Comment by samuelcolvin** [comment]
Has this changed recently @dromer?

**Comment by dromer** [comment]
I'm only using the plugin since this week and this is the only highlighting I've seen.

The main problem that I see is that any single set of `{ }` is being identified as belonging to the `.j2` syntax and anything inside of it will subsequently be highlighted incorrectly. Instead it should only look for `{% %}` and `{{ }}` enclosed code.
(I understand why the later doesn't work inside `" "`, but that's a separate issue I assume)

Sorry if I can't give any more info. I can link to these files if you need them.

**Comment by samuelcolvin** [comment]
humm, have you tried disabling `bracketPairColorization`?

**Comment by dromer** [comment]
Then the jinja2 highlighting is completely broken:
![](https://i.imgur.com/CNl0yti.png)

(looks exactly the same as if the file just had `.cpp` extension)

**Comment by RedCMD** [comment]
I think you confuse `bracketPairColorization` with syntax highlighting 
the bright colored bracket highlighting is handled separately by vscode
and is the same for all languages
the extension just tells vscode what type of brackets exist

if you try just the plain cpp language with `bracketPairColorization` enabled
all cpp brackets will be highlighted

other than `{{class_name}}`
it doesn't seem like any of jinja2 syntax highlighting is working

**Comment by dromer** [comment]
@RedCMD indeed, thnx for the clarification.  Still it seems that the jinja2 scope isn't recognized when enclosed in a pair of single brackets. At least that's what I think we're observing here.
